### DISCUSSION TITLE
remark forms plugin

### DISCUSSION THREAD

**Discussion by shriharip** [remarkjs/remark#1257] [Show and tell]
Just wanted to share the a plugin I made.  Markdown syntatic sugar to generate HTML forms. 
You can find the repo here at  [remark-forms](https://github.com/shriharip/remark-forms) .
would be great if somebody would love to test it out.

**Comment by ChristianMurphy** [comment]
Welcome @shriharip! 👋 
Thanks for sharing! It looks interesting.

Some thoughts and ideas on the code structure.

Code Convention:
* Consider using a type checker like TypeScript for JSDoc to help catch issues around data types (https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html)
* Consider using a linter like `xo` (https://github.com/xojs/xo) or ESLint (https://eslint.org/) to look for code smells and other potential issues

Code Style:
* Consider using a formatter like prettier (https://prettier.io/) to ensure consistent code style

Testing:
* Consider using `node:test` over `mocha`. `node:test` is faster, a native part of node, and has a similar API. (https://nodejs.org/api/test.html)
* Consider leveraging `node:assert` over `expect.js`. `node:assert` is a native part of Node and is still fairly easy to read and follow. 
* Consider using GitHub Actions to automatically run tests across different versions of Node.js (https://docs.github.com/en/actions)

Dependency management:
* Consider leveraging `npm` as the package manager. It is the most commonly used JavaScript package manager, and is fairly fast these days
* Consider moving TypeScript from `dependencies` to `devDependencies`, it is a build time tool and is not needed at runtime (https://github.com/shriharip/remark-forms/blob/90d220da896dd84a6482fbdc1802fd2794f80df7/package.json#L24)
* Consider dropping `parcel` entirely. Libraries should generally ship raw source, rather than code bundles (https://github.com/shriharip/remark-forms/blob/90d220da896dd84a6482fbdc1802fd2794f80df7/package.json#L33)
* Consider fully leveraging `hastscript` for most/all `hast` node creation (https://github.com/shriharip/remark-forms/blob/90d220da896dd84a6482fbdc1802fd2794f80df7/index.js#L195-L211) or dropping it if you do not plan on using it.

**Reply by shriharip** [reply]
Hey @ChristianMurphy . Thanks for the feedback. Will start looking into these.
